server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -290,24 +316,138 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(final ServerCall<Re
+                                log.info().append(serviceName).append(".").append(methodName)
+                                        .append(": request unauthenticated").endl();
+                            } else if (status.equals(Status.Code.PERMISSION_DENIED)) {
+                                log.info().append(serviceName).append(".").append(methodName)
+                                        .append(": permission denied ").endl();
+                            } else {
+                                log.error().append(cause).endl();
+                            }
+
+                            call.close(sre.getStatus(), sre.getTrailers());
+                            return new ServerCall.Listener<>() {};
+                        } else {
+                            log.error().append("Failed to invoke auth method: ").append(e).endl();
+                            return onFailure.get();
+                        }
+                    }
+                }
+
+                final ServerCall.Listener<ReqT> delegate = serverCallHandler.startCall(innerCall, headers);
+                return new ForwardingServerCallListener.SimpleForwardingServerCallListener<>(delegate) {
+                    @Override
+                    public void onMessage(ReqT message) {
+                        // authorize requests now if the client was required to send a single request
+                        if (!isClientStreaming && authMethod != null) {
+                            try {
+                                authMethod.invoke(authWiring, authContext, message);
+                            } catch (IllegalAccessException | InvocationTargetException e) {
+                                final Throwable cause = e.getCause();
+                                if (cause instanceof StatusRuntimeException) {
+                                    final StatusRuntimeException sre = (StatusRuntimeException) cause;
+                                    Status.Code status = sre.getStatus().getCode();
+                                    if (status.equals(Status.Code.UNAUTHENTICATED)) {
+                                        log.info().append(serviceName).append(".").append(methodName)
+                                                .append(": request unauthenticated").endl();
+                                    } else if (status.equals(Status.Code.PERMISSION_DENIED)) {
+                                        log.info().append(serviceName).append(".").append(methodName)
+                                                .append(": permission denied ").endl();
+                                    } else {
+                                        log.error().append(cause).endl();
+                                    }
+
+                                    call.close(sre.getStatus(), sre.getTrailers());
+                                } else {
+                                    log.error().append("Failed to invoke auth method: ").append(e).endl();
```

Comment:
I know we have places that take different approaches, but I think we might want to encourage `log.error(e).append("Failed to invoke auth method").endl()` approach instead. (Probably deserves an audit of our current usages and an effort to consolidate our approaches.)